### ISSUE TITLE
Add methods AddWithData(str string, data interface{}) and Data(line int) interface{} to Choice

### ISSUE THREAD

**Issue by diegoefe** [pwiecz/go-fltk#107]
Is useful to avoid when using db ids for one to many relationships.
i.e.: table  car_brands (id int, brand text). In the Choice we show the brand but not the id, but we need it when updating related table

**Comment by pwiecz** [comment]
I'm not sure it's really needed.
 In Go it's easy to pass capture the data in the callback passed to the Add/Insert method.
It's even more type-safe and doesn't require casting back from interface{} to a concrete type.
It may be a bit more verbose as instead of `choice.Add(label1, id1, onSelected)` you need to write `choice.Add(label1, func() { onSelected(id1) })`, but it seems both type-safe and adaptable to one's needs...

**Comment by diegoefe** [comment]
Ok, sounds good, if i'm understanding you correctly, is that we need to store at least the id in the widget

**Comment by pwiecz** [comment]
I wonder why you need to store the id inside the widget if (as far as I understand) you only need to use it as a parameter to the callback?
You can just capture it in the callback passed to the Choice.

**Comment by diegoefe** [comment]
Maybe i don't know how to do it...
```go
types := []Type{}
db := DBInstance()
db.Model(&Type{}).Order("name").Find(&types)
ttype.Add("Seleccione...", nil)
for _, t := range types {
	ttype.Add(t.Name, func() {
		// FIXME/TODO: resolver como guardar los id's
		fmt.Printf("Selected [%s] [%d]\n", t.Name, t.Id)
	})
}
ttype.SetValue(0)
```
This always prints:
Selected [LastLabel] [2]
Selected [LastLabel] [2]
Selected [LastLabel] [2]

In C++ I use user_data(...) and user_data()(cast....)

**Comment by pwiecz** [comment]
Oh, you've hit one of those Golang quirks that's going to be "fixed" in the next Golang version (1.22) - you can read about it here: https://go.dev/blog/loopvar-preview
To make it work, you have to change the loop to something similar to:
```
for _, t := range types {
        tt := t
	ttype.Add(t.Name, func() {
		// FIXME/TODO: resolver como guardar los id's
		fmt.Printf("Selected [%s] [%d]\n", tt.Name, tt.Id)
	})
}
```

**Comment by diegoefe** [comment]
Ok, I've learn something today. Thank you very much!
I guess I can go by with this. Thanks!
